### PR TITLE
Use argparse

### LINKED ISSUE #1: Move to argparse
We could possibly move to argparse since Lithium is still using getopt, which should use less code.

https://docs.python.org/2/library/getopt.html
https://docs.python.org/2/library/argparse.html

### PR DESCRIPTION
**Pull request by nth10sd**
@tysmith moved lithium to use argparse, I fixed up a thing or two here and there. We will preserve the naming of tmp directories to use tmp1, tmp2, etc. though.

This fixes #1 .

### CODE REVIEW

File: lithium/lithium.py [hunk 1]
```diff
@@ -162,46 +89,108 @@ def main():
...
+    minimizeChunkStart = args.chunkstart
+    minimizeRepeatFirstRound = args.repeatfirstround
+    if args.maxruntime:
+        stopAfterTime = time.time() + args.maxruntime
+    extra_args = args.extra_args[0]
+
+    if args.testcase is None:
+        try:
+            testcaseFilename = extra_args[-1] # can be overridden by --testcase in processOptions
+        except IndexError:
```
Comment (jruderman) [inline lithium/lithium.py]: I prefer avoiding "try/except" in favor of e.g. a length check

File: lithium/lithium.py [hunk 2]
```diff
@@ -162,46 +89,108 @@ def main():
...
+            parser.print_help()
+            return False
+    else:
+        testcaseFilename = args.testcase
+    testcaseExtension = os.path.splitext(testcaseFilename)[1]
+
+    if not isPowerOfTwo(minimizeMin) or not isPowerOfTwo(minimizeMax) or len(extra_args) < 1:
```
Comment (jruderman) [inline lithium/lithium.py]: Please restore the old error messages for the 'not isPowerOfTwo' cases. The 'len(extra_args) < 1' case should already be handled above.

**Comment by nth10sd** [comment]
I squashed the follow-up commit with @jruderman's suggested changes, following this [suggested command in Stack Overflow](http://stackoverflow.com/a/16642628) and doing a forced push for the PR.

@tysmith does the PR look good to you, esp. the lines around [this location](https://github.com/MozillaSecurity/lithium/pull/3/commits/0e3dd3dea8476d3742266549646fa31210209fbb#diff-6c94e88b988f8496ba8bc2c2d6308233R176)?

**Comment by tysmith** [comment]
Looks good to me. If you have tested it and you are confident it works as expected feel free to merge.

**Comment by nth10sd** [comment]
Note that the change to use tempfile was removed.

I had done some tests locally and it seems fine. We'll deal with future bugs as they come along.

**Pull request merged**